### PR TITLE
Add deprecation warning for Rails < 5.2

### PR DESCRIPTION
**Pull request by mjobin-mdsol**
What do you think of officially deprecating Rails `< 5.2` in the next release?
we start by adding a Warning for 0.12.0
let that go for a while.
and then modify the gemspec for 0.13.0 or update the numbering? 0.20.0 ? 1.0 ?

### CODE REVIEW

**Review by jphenow — APPROVED** [review]
Yea I'm fine with that I think.

**Review by Zogoo — APPROVED** [review]
Yeah, since ruby 2.4 dropped from the gem. It's a good idea to not support Rails under 5 version.

**Comment by mjobin-mdsol** [comment]
we can release 0.13 with this, or close this and go straight with #144

**Comment by jphenow** [comment]
let's go with the other I think, I'd be concerned for folks on <5.2 since Rails itself is barely patching below that.

**Comment by jphenow** [comment]
Just remembered we let folks select their rails so I think we want _both_ actually.

**Review by jphenow — APPROVED** [review]

**Comment by jphenow** [comment]
once travis is green I'll merge this and #114 and I can cut a new version sometime today hopefully

**Comment by mjobin-mdsol** [comment]
@jphenow I don't think you need both
if you merge #114 the gem can't be installed with rails below 5.2, and thus this warning is useless.
although I opened this, I am now in favor of skipping this change and just bumping the requirement to 5.2

5.1 and below are no longer maintained by the Rails core team

**Comment by jphenow** [comment]
Rails is only ever a development dependency technically because the gem doesn't require it be used with Rails. So generally both seem fair.

**Comment by mjobin-mdsol** [comment]
ah, makes sense. up to you then.

**Comment by mjobin-mdsol** [comment]
i just double check and `activesupport` is a runtime dependency. and I am bumping it as well.
so, after #144 is merged, 5.1.x and below cannot be used with this gem.
unless there is still something I am missing?

**Comment by mjobin-mdsol** [comment]
I confirm, after merging #144 
its no longer possible to install the gem in 5.1, making this warning useless

```
Bundler could not find compatible versions for gem "activesupport":
  In snapshot (Gemfile.lock):
    activesupport (= 5.1.7)

  In Gemfile:
    saml_idp was resolved to 0.12.0, which depends on
      activesupport (>= 5.2)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by Zogoo** [comment]
@mjobin-mdsol in that case you may want to close this pull request?

**Comment by mjobin-mdsol** [comment]
yes

**Pull request closed without merging**
